File: Reflow/ILinkWriter.cs
using Reflow.Tasks;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace Reflow
{
    public interface ILinkWriter
    {
        void Write(IDataReader source, string table, TaskExecutionContext context);
        void Initialize(IDataLink link );

    }
}

File: Java/src/main/java/com/nuarca/etl/provider/ILinkWriter.java
package com.nuarca.etl.provider;

import com.nuarca.etl.tasks.TaskExecutionContext;

import java.sql.ResultSet;

public interface ILinkWriter {

    void write(ResultSet source, String table, TaskExecutionContext context) throws Exception ;

    void initialize(IDataLink link) throws Exception ;
}
